numpy/lib/arraypad.py

```diff
@@ -997,52 +996,75 @@ def _normalize_shape(narray, shape):
     ----------
     narray : ndarray
         Input ndarray
-    shape : {sequence, int}, optional
-        The width of padding (pad_width) or the number of elements on the
-        edge of the narray used for statistics (stat_length).
+    shape : {sequence, array_like, float, int}, optional
```

Comment:
I'm not super-worried either way b/c this function is an internal implementation detail, but maybe the right thing is just to rename the function and argument? like make it `_normalize_edge_argument(ndarray, edge_argument, cast_to_int=True)`? AFAICT the functionality is fine, all the different things that get passed in really do want the same handling, it's just that that handling is not really normalizing a "shape".